comp/metadata/host/hostimpl/utils/meta.go

```diff
@@ -74,6 +75,7 @@ func getMeta(ctx context.Context, conf model.Reader, hostnameComp hostnameinterf
 		InstanceID:                instanceID,
 		AgentHostname:             agentHostname,
 		HostnameResolutionVersion: 1,
+		CCRID:                     cloudproviders.GetCCRID(ctx),
```

Comment:
If you look at how `GetHostAliases` works, it does extra work to be fast because this code runs at Agent startup. `GetHostAliases` sends requests to all possible metadata endpoints at once, rather than iterating over them and blocking on each. In order to not do this twice, [my version](https://github.com/DataDog/datadog-agent/pull/39474/files) of this code changes `GetHostAliases` to return both the aliases and the cloudname, and then `GetCCRID` uses that cloudname to only lookup the CCRID for that exact cloudenv. What do you think?